src/main/functions/DIDResolution.ts

```diff
@@ -31,6 +31,63 @@ export function getResolver(opts: ResolveOpts): Resolvable {
   }
 }
 
+/**
+ * This method returns a resolver object in OP/RP
+ * If the user of this library, configures OP/RP to have a customResolver, we will use that
+ * If the user of this library configures OP/RP to use a custom resolver for any specific did method, we will use that
+ * and in the end for the rest of the did methods, configured either with calling `addDidMethod` upon building OP/RP
+ * (without any resolver configuration) or declaring in the subject_syntax_types_supported of the registration object
+ * we will use universal resolver from Sphereon's DID Universal Resolver library
+ * @param customResolver
+ * @param subjectSyntaxTypesSupported
+ * @param resolverMap
+ */
+export function getResolverUnion(
+  customResolver: Resolvable,
+  subjectSyntaxTypesSupported: string[] | string,
+  resolverMap: Map<string, Resolvable>
+): Resolvable {
+  if (customResolver) {
+    return customResolver;
+  }
+  const fallbackResolver: Resolvable = customResolver ? customResolver : new UniResolver();
+  const uniResolvers: {
+    [p: string]: (did: string, _parsed: ParsedDID, _didResolver: Resolver, _options: DIDResolutionOptions) => Promise<DIDResolutionResult>;
+  }[] = [];
+  const subjectTypes: string[] = [];
+  if (subjectSyntaxTypesSupported) {
+    typeof subjectSyntaxTypesSupported === 'string'
+      ? subjectTypes.push(subjectSyntaxTypesSupported)
+      : subjectTypes.push(...subjectSyntaxTypesSupported);
+  }
+  if (subjectTypes.indexOf(SubjectSyntaxTypesSupportedValues.DID.valueOf()) !== -1) {
+    return customResolver ? customResolver : new UniResolver();
+  }
+  const specificDidMethods = subjectTypes.filter((sst) => !!sst && sst.startsWith('did:'));
+  specificDidMethods.forEach((dm) => {
+    let uniResolver;
+    if (!resolverMap.has(dm)) {
+      uniResolver = getUniResolver(getMethodFromDid(dm));
+    } else {
+      uniResolver = getUniResolver(getMethodFromDid(dm), resolverMap.get(dm));
```

Comment:
Why are we always using a uni resolver?

Reply:
we're using UniResolver with the config provided from the user

Reply:
But a difResolver is not a uni resolver. The uni resolver delegates to something hosted, whilst a dif resolver could be resolving directly for instance.

Reply:
changed it to:
```
 if (!resolverMap.has(dm) || resolverMap.get(dm) === null) {
      methodResolver = getUniResolver(getMethodFromDid(dm));
    } else {
      methodResolver = resolverMap.get(dm);
    }
```